molgenis-data/src/main/java/org/molgenis/util/EntityUtils.java

```diff
@@ -356,10 +360,39 @@ public static boolean equals(AttributeMetaData attr, AttributeMetaData otherAttr
 
 		// identifier might be null if attribute hasn't been persisted yet
 		if (otherAttr.getIdentifier() != null)
-		{
 			if (!Objects.equals(attr.getIdentifier(), otherAttr.getIdentifier())) return false;
+		return compareAttributeAttributes(attr, otherAttr);
+	}
+
+	/**
+	 * Returns true if an attribute equals another attribute.
+	 * Includes the identifier, even if the identifier is null
+	 *
+	 * @param attr
+	 * @param otherAttr
+	 * @return
+	 */
+	public static boolean equals(AttributeMetaData attr, AttributeMetaData otherAttr)
+	{
+		if (attr == null || otherAttr == null)
+		{
+			if (attr == null && otherAttr == null) return true;
```

Comment:
remove this: if (attr == null && otherAttr == null) return true;
 if (attr == null || otherAttr == null) return false is enough